demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs

```diff
@@ -301,4 +314,36 @@ @@@@@@@@@@@         @@@@@@@@@@@@@
         Console.ResetColor();
         Console.WriteLine();
     }
+
+    //HACK: Needed until CoreFX implements Environment.OSVersion
+    private static string DetectOS()
+    {
+        Process proc = new Process();
+
+        string detectionResult = String.Empty;
+
+        if (proc.StartInfo.Environment.ContainsKey("TERM_PROGRAM"))
```

Comment:
Any reason not to use Environment.GetEnvironmentVariable rather than using Process?